File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // int n = nums.size();
        // int i = 0,j = n-1;
        // //sort(nums.begin(),nums.end());
        // while(i < j){
        //     int sum = nums[i]+nums[j];
        //     if(sum == target) return{i,j};
        //     else if(sum > target) j--;
        //     else i++;
        // }
        // return {};

          unordered_map<int, int> mp;

        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];

            if (mp.find(complement) != mp.end()) {
                return {mp[complement], i};
            }

            mp[nums[i]] = i;
        }

        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # n = len(nums)
        # for i in range(n):
        #     for j in range(i+1,n):
        #         if nums[i]+nums[j] == target:
        #             return[i,j]
        
        # -------second approach -------
        # n = len(nums)
        seen = {}
        for i,num in enumerate(nums):
            comp = target - num
            if comp in seen:
                return [seen[comp],i]
            seen[num] = i